apps/backend/src/libs/plugins/authorization/authorization.plugin.ts

```diff
@@ -0,0 +1,69 @@
+import { type FastifyRequest } from "fastify";
+import fp from "fastify-plugin";
+
+import { ExceptionMessage } from "~/libs/enums/enums.js";
+import { HTTPCode } from "~/libs/modules/http/http.js";
+import { type Token } from "~/libs/modules/token/token.js";
+import { AuthError } from "~/modules/auth/auth.js";
+import { type UserService } from "~/modules/users/user.service.js";
+
+import { WHITE_ROUTES } from "../libs/constants/constants.js";
+import { isTokenExpiredError } from "../libs/helpers/helpers.js";
+
+type AuthorizationPluginOptions = {
+	token: Token;
+	userService: UserService;
+};
+
+const authorization = fp<AuthorizationPluginOptions>(
+	(fastify, options, done) => {
+		const { token, userService } = options;
+
+		fastify.decorateRequest("user", null);
+
+		fastify.addHook("preHandler", async (request: FastifyRequest) => {
+			const { url } = request.raw;
+
+			for (const whiteRoute of WHITE_ROUTES) {
```

Comment:
whiteRoutes should be passed as a parameter from the server-application. And we also should keep WHITE_ROUTES constant in `/src/libs/modules/server-application/libs/constants/white-routes.constant.ts`